### PR TITLE
Add local storage for the form data.

### PR DESCRIPTION
**Pull request by taher-software**
In this Branch, you can find the following changes :100: 
- create object with all the data from  and save it in local storage.
- Add events to track changes in any input field.

### CODE REVIEW

File: js/window.js [hunk 1]
```diff
@@ -206,4 +206,30 @@ form.addEventListener('submit', (event) => {
...
\ No newline at end of file
+});
+/* save form data */
+const userName = form.elements.name;
+const userMail = form.elements.email;
+const userMessage = form.elements.message;
+function populateStorage() {
+  localStorage.setItem('name', form.elements.name.value);
+  localStorage.setItem('mail', form.elements.email.value);
+  localStorage.setItem('message', form.elements.message.value);
```
Comment (icebox827) [inline js/window.js:217]: - [ ] **REQUIRED:** Please, in order to be saved to local storage the data object must be turned into a string using `JSON.stringfy`

File: js/window.js [hunk 2]
```diff
@@ -206,4 +206,30 @@ form.addEventListener('submit', (event) => {
...
+}
+function setForm() {
+  const currentUserName = localStorage.getItem('name');
+  const currentUserEmail = localStorage.getItem('mail');
```
Comment (icebox827) [inline js/window.js:221]: - [ ] **REQUIRED:** Please, in order to be used after retrieve from the local storage the data object must be turned back into an object using `JSON.parse`

**Review by icebox827 — CHANGES_REQUESTED** [review]
Hi Taher, 

Good job so far!
Your project looks good, but there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/window.js [hunk 3]
```diff
@@ -206,4 +206,31 @@ form.addEventListener('submit', (event) => {
...
   }
-});
\ No newline at end of file
+});
+/* save form data */
+const userName = form.elements.name;
+const userEmail = form.elements.email;
+const userMessage = form.elements.message;
+function populateStorage() {
+  const userInput = [form.elements.name.value, form.elements.email.value,
```
Comment (Mihndim2020) [inline js/window.js:215]: - [ ] As requested by the previous code reviewer, which also a requirement for this project, please make sure you create a single JavaScript object with all the data from the entire form and save it in local storage. You have an array of inputs, please kindly change that. Please this [article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Working_with_Objects) will be of help. 
- [ ] Your data is not been saved to the local storage, when you start filling the form and refresh(reload) the page, the data is lost, meanwhile it is supposed to be saved to the local storage, please add a change event listener to the form to track the changes in the input fields and save the data to local storage. Checkout this [article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Working_with_Objects).


tahir22

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
## Hi @Taher-web-dev & @icebox827 , 

Great work on making the changes requested by a previous reviewer 👏 👏 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
stay-strong-youre-almost-there


 # Status: Requires Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @Taher-web-dev, 

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**